crates/compilers/src/preprocessor/mod.rs

```diff
@@ -0,0 +1,261 @@
+            // Set remappings into HIR parsing context.
+            for remapping in &paths.remappings {
+                parsing_context
+                    .file_resolver
+                    .add_import_map(PathBuf::from(&remapping.name), PathBuf::from(&remapping.path));
+            }
+            // Load and parse test and script contracts only (dependencies are automatically
+            // resolved).
+            let preprocessed_paths = sources
+                .into_iter()
+                .filter(|(path, source)| {
+                    is_test_or_script(path, paths) && !source.content.is_empty()
+                })
+                .map(|(path, _)| path.clone())
+                .collect_vec();
+            parsing_context.load_files(&preprocessed_paths)?;
+
+            let hir_arena = ThreadLocal::new();
+            if let Some(gcx) = parsing_context.parse_and_lower(&hir_arena)? {
+                let hir = &gcx.get().hir;
+                // Collect tests and scripts dependencies and identify mock contracts.
+                let deps = PreprocessorDependencies::new(
+                    &sess,
+                    hir,
+                    &preprocessed_paths,
+                    &paths.paths_relative().sources,
+                    &paths.root,
+                    mocks,
+                );
+                // Collect data of source contracts referenced in tests and scripts.
+                let data = collect_preprocessor_data(&sess, hir, &deps.referenced_contracts);
+
+                // Extend existing sources with preprocessor deploy helper sources.
+                sources.extend(create_deploy_helpers(&data));
+
+                // Generate and apply preprocessor source updates.
+                apply_updates(sources, remove_bytecode_dependencies(hir, &deps, &data));
+            }
+
+            Ok(())
+        });
+
+        // Return if any diagnostics emitted during content parsing.
+        if let Err(err) = sess.emitted_errors().unwrap() {
+            trace!("failed preprocessing {err}");
+            return Err(SolcError::Message(err.to_string()));
+        }
+
+        Ok(())
+    }
+}
+
+impl Preprocessor<MultiCompiler> for TestOptimizerPreprocessor {
+    fn preprocess(
+        &self,
+        compiler: &MultiCompiler,
+        input: &mut <MultiCompiler as Compiler>::Input,
+        paths: &ProjectPathsConfig<MultiCompilerLanguage>,
+        mocks: &mut HashSet<PathBuf>,
+    ) -> Result<()> {
+        // Preprocess only Solc compilers.
+        let MultiCompilerInput::Solc(input) = input else { return Ok(()) };
+
+        let Some(solc) = &compiler.solc else { return Ok(()) };
+
+        let paths = paths.clone().with_language::<SolcLanguage>();
+        self.preprocess(solc, input, &paths, mocks)
+    }
+}
+
+/// Helper function to compute hash of [`interface_representation`] of the source.
+pub(crate) fn interface_representation_hash(source: &Source, file: &PathBuf) -> String {
```

Comment:
&PathBuf -> &Path

Reply:
changed in https://github.com/foundry-rs/compilers/pull/252/commits/64c5ecdc8572b6eef410e3385d29dbb0f61c7558